Teensy/src/gpio/gpio.cpp

```diff
@@ -96,7 +96,35 @@ static GyroData_s gpio_getGyro()
 
 static daq_BatteryV_t gpio_getVBat()
 {
-    return 0U;
+    #include <Arduino.h>
+    // error handling
+    #define NO_STEPPED_DOWN_VOLTAGE 100
+
+    // A voltage divider should be scaling down the voltage from 12V to 2.4 V before this
+    // refer to https://www.instructables.com/Voltage-Measurement-Using-Arduino/ for how this formula works
+    // the formula for 2 resistors connected in series, forming the voltage divider is: V1 = Vm * (R2/(R1+R2))
+
+    // error handling for A16 returning NULL
+    if(analogRead(PIN_A16) == NULL){
```

Comment:
The API isnt clear but I having quick skim over the source code I think analogRead() returns zero if an error was to occur. So this error handling can be removed. Also just to let you know, in embedded systems exit is not used bc theres not task manager like there in OS. Generally things like this are just logged as error messages

Reply:
Okay thanks for the info! Will take note 😬